src/test/regress/expected/subselect_gp_optimizer.out

```diff
@@ -509,7 +509,6 @@ explain SELECT * FROM csq_r WHERE exists (SELECT * FROM csq_f(csq_r.a));
          SubPlan 1  (slice1; segments: 3)
            ->  Result  (cost=0.00..0.00 rows=1 width=1)
                  ->  Result  (cost=0.00..0.00 rows=1 width=1)
-                       ->  Result  (cost=0.00..0.00 rows=1 width=1)
```

Comment:
How the new code affects this node?

Reply:
verbose shows that patch removes redundant node:
before
```sql
explain (verbose, costs off)
SELECT * FROM csq_r WHERE exists (SELECT * FROM csq_f(csq_r.a));
                QUERY PLAN                
------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: a
   ->  Seq Scan on postgres.csq_r
         Output: a
         Filter: (SubPlan 1)
         SubPlan 1  (slice1; segments: 3)
           ->  Result
                 Output: 
                 ->  Result
                       Output: csq_r.a, 
                       ->  Result
                             Output: true
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(14 rows)

explain (verbose, costs off)
SELECT * FROM csq_r WHERE not exists (SELECT * FROM csq_f(csq_r.a));
                QUERY PLAN                
------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: a
   ->  Seq Scan on postgres.csq_r
         Output: a
         Filter: (SubPlan 1)
         SubPlan 1  (slice1; segments: 3)
           ->  Result
                 Output: 
                 ->  Result
                       Output: csq_r.a, 
                       ->  Result
                             Output: true
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(14 rows)
```
after
```sql
explain (verbose, costs off)
SELECT * FROM csq_r WHERE exists (SELECT * FROM csq_f(csq_r.a));
                QUERY PLAN                
------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: a
   ->  Seq Scan on postgres.csq_r
         Output: a
         Filter: (SubPlan 1)
         SubPlan 1  (slice1; segments: 3)
           ->  Result
                 Output: csq_r.a
                 ->  Result
                       Output: true
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(12 rows)

explain (verbose, costs off)
SELECT * FROM csq_r WHERE not exists (SELECT * FROM csq_f(csq_r.a));
                QUERY PLAN                
------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: a
   ->  Seq Scan on postgres.csq_r
         Output: a
         Filter: (SubPlan 1)
         SubPlan 1  (slice1; segments: 3)
           ->  Result
                 Output: csq_r.a
                 ->  Result
                       Output: true
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(12 rows)
```

Reply:
change rempved